### ISSUE TITLE
Stripping types should include ts-ignore comments

### ISSUE THREAD

**Issue by ljharb** [microsoft/TypeScript#37144]
**TypeScript Version:**  3.8 latest, playground

**Search Terms:** remove ts-ignore comment

**Code**

```ts
// this is a comment that should be kept
const message: string = 'hello world';
// the below ts-ignore comment should be removed, since it has no meaning in a non-TS context
// @ts-ignore
console.log(message);
```

**Expected behavior:**
ts-ignore comment is stripped, other comments are kept.

**Actual behavior:**
all comments are kept including ts-ignore comments.

**Playground Link:** https://www.typescriptlang.org/play/#code/PTAEBcAsEsGdTqAhqAxgewLaYKYDtwJIlDZJ0BXAGwBNQAjHUAaxwAdwAoDPWQ3WLCQBzHAC5QfAE7Q8w0AF5QAckg4qVdKADu6KbWUBuTiCJNGm7RFgBaaMLx6mGbPlLlqJsAAFwt+45SONzovOhUOAB0msIAFAJCogCUhkA

**Related Issues:**

**Comment by tadhgmister** [comment]
I do want to point out that `@ts-ignore` doesn't have to be on it's own line, you can write this

```ts
declare function doThingWithNumber(val:number): void
// this is a comment that should be kept
const message: string = 'hello world';
// @ts-ignore because this isn't expected to fail at runtime ever but might.
doThingWithNumber(message);
```

I would definitely not want the whole line to be removed, I'd kind of argue that just keeping the whole line makes a bit more sense than modifying the comment to only remove the @ts-ignore, especially since other compiler orders like `@eslint-disable-next-line` are still kept.

**Comment by ljharb** [comment]
I don’t agree; ts-ignore is a directive to tsc, and once tsc has run, it’s irrelevant. The extra context is for the *ts-ignore* and is also irrelevant once emitted.

**Comment by ljharb** [comment]
Eslint commands are for eslint; those are none of typescript’s business.

**Comment by fasttime** [comment]
I was actually surprised to find `@ts-expect-error` comments in my generated .js files. Like `@ts-ignore`, these are pointless outside TypeScript code.

**Comment by rflechtner** [comment]
> I was actually surprised to find `@ts-expect-error` comments in my generated .js files. Like `@ts-ignore`, these are pointless outside TypeScript code.

I was even running into issues where explicitly escaping a type error with `@ts-expect-error` was compiled to js code where that type error was no longer detected, failing our tests (because the expected error did not occur).

**Comment by haysclark** [comment]
+1 to having an option to remove `// @ts-ignore`.

I'm using TypeScript to generate Adobe-friendly JS and ExtendScript which is really picky about comments with `@` symbols.
